pkg/security/probe/model.go

```diff
@@ -66,10 +67,15 @@ func (m *Model) NewEvent() eval.Event {
 func (m *Model) ValidateField(key string, field eval.FieldValue) error {
 	// check that all path are absolute
 	if strings.HasSuffix(key, "filename") || strings.HasSuffix(key, "_path") {
-		value, ok := field.Value.(string)
-		if ok {
-			if value != path.Clean(value) || !path.IsAbs(value) {
-				return fmt.Errorf("invalid path `%s`, all the path have to be absolute", value)
+		if value, ok := field.Value.(string); ok {
+			errAbs := fmt.Errorf("invalid path `%s`, all the path have to be absolute", value)
+
+			if matched, err := regexp.Match(`\.\.`, []byte(value)); err != nil || matched {
+				return errAbs
+			}
```

Comment:
Should we add a test for `./` or `/.` and `//` patterns as well ? I'm specifically thinking about `*/` and also `/etc//.//././//////././/` which is still a valid path, and I'm not sure how these would be handled.

Reply:
Alternatively we could simply use `filepath.Clean()` and check if the input is equal to the output at the `patternToRegexp` stage see below.

Reply:
I re-added the `path.Clean` check an added more tests. We can move the check in the  `patternToRegexp` since this function can be used for other fields than `path` or `filename`